### PR TITLE
feat(queryhandler): add built-in pierce handler

### LINKED ISSUE #6217: Shadow Dom support
Is there any support for finding elements in the Shadow Dom? Our application uses lit-html components. Please provide some examples.

### PR DESCRIPTION
**Pull request by johanbay**
Adds a handler 'pierce' that pierces shadow roots while querying.
Closes #6217.

### CODE REVIEW

File: src/common/QueryHandler.ts [hunk 1]
```diff
@@ -112,7 +112,57 @@ const _defaultHandler = makeQueryHandler({
...
+    };
+    if (element instanceof Document) {
+      element = element.documentElement;
+    }
+    collect(element);
+    return result;
+  },
+});
+
+const _builtInHandlers = new Map([['aria', ariaHandler], ['pierce', pierceHandler]]);
```
Comment (mathiasbynens) [inline src/common/QueryHandler.ts]: Now that there are multiple entries, let's split this across multiple lines for readability:

```suggestion
const _builtInHandlers = new Map([
  ['aria', ariaHandler],
  ['pierce', pierceHandler],
]);
```

(unless Prettier likes to put it all on one line for some reason?)

**Review by mathiasbynens — APPROVED** [review]

**Comment by php4fan** [comment]
Why is this still undocumented??

**Comment by j-wsl** [comment]
is it possible to get this with xpaths?

**Comment by gitressa** [comment]
I was also looking for this and a simple example, and found one by @Remco Vaes in https://stackoverflow.com/a/76967459:

> This issue is solved with [P selectors in Puppetteer](https://pptr.dev/guides/query-selectors/#-and--combinators).
> 
> In this case you could use `page.click('>>> #my-button')`. The `>>>` Allow you to select stuff in shadowRoots.
> 
> I know this issue is old, but I needed this and this issue still pops up on searches.

**Pull request merged**